### ISSUE TITLE
Show detailed error messages

### ISSUE THREAD

**Issue by chao2zhang** [detekt/detekt#3423]
## Expected Behavior
When a final newline is reported, the error message will be shown in the console:
```
/Users/cazhang/detekt/detekt-formatting/src/main/kotlin/io/gitlab/arturbosch/detekt/formatting/FormattingRule.kt:1:1: File must end with a newline (\n)
```

## Current Behavior
The error message is shown in the console
```
formatting - 5min debt
        FinalNewline - [] at /Users/cazhang/detekt/detekt-formatting/src/main/kotlin/io/gitlab/arturbosch/detekt/formatting/FormattingRule.kt:1:1
```

## Context
I have received reports from other devs complaining they don't know what FinalNewLine is (Probably related to the line and column, but this is a fix for upstream(ktlint). However, I found that ktlint shows detailed error messages which help people to understand the issue.

**Comment by schalkms** [comment]
I'm not sure whether this should be done for the console report. The console report could become quite verbose.

**Comment by chao2zhang** [comment]
We can provide an option to specify the verbosity of output. The issue description and message description are both shown in the SARIF report output (See https://github.com/chao2zhang/detekt/pull/4/files) and most of the rules are currently displaying the same message as the issue description

**Comment by schalkms** [comment]
Yes, that’s true. I always root for having the console and XML report to display concise information. For more detailed information there’s the HTML or SARIF report available.

**Comment by schalkms** [comment]
@cortinico & @BraisGabin what's your take on this? I think we should either close this issue or tag it as improvement depending on the approved decision.

**Comment by cortinico** [comment]
> I think we should either close this issue or tag it as improvement

Potentially this

I'm curious to know what's the message we receive from ktlint here:
https://github.com/detekt/detekt/blob/f5b6b04b0822b670b206015c6cf91706e73467b1/detekt-formatting/src/main/kotlin/io/gitlab/arturbosch/detekt/formatting/FormattingRule.kt#L89

Isn't that `File must end with a newline (\n)`?

**Comment by cortinico** [comment]
> @cortinico & @BraisGabin what's your take on this? I think we should either close this issue or tag it as improvement depending on the approved decision.

Closing as the agreement seems to be leaning towards that direction. Users can still enable other type of reports + some of the rules are reporting long messages (which is good for HTML report and so on), but will end up polluting the console.
